python/GafferSceneUI/SetUI.py

```diff
@@ -221,15 +234,19 @@ def __setsPopupMenu( menuDefinition, plugValueWidget ) :
 		return
 
 	with plugValueWidget.getContext() :
-		if acceptsSetNames :
-			currentNames = set( plug.getValue().split() )
-		elif acceptsSetName :
-			currentNames = set( [ plug.getValue() ] )
-		else :
-			currentExpression = plug.getValue()
+		plugValue = plug.getValue()
+
+	if acceptsSetNames :
+		currentNames = set( plugValue.split() )
+	elif acceptsSetName :
+		currentNames = set( [ plugValue ] )
+	else :
+		currentExpression = plugValue
+
+	textSelection = plugValueWidget.textWidget().getSelection()
+	selectionSetExpression = plugValue[textSelection[0]: textSelection[1]] if textSelection != ( 0, 0 ) else plugValue
```

Comment:
Fixed in 82b0f5fa1764c5356c50f8d61b53bf7b2240d06d. And since I have `plugValue` now, there's not much point in duplicating it as `currentExpression`, so I removed `currentExpression` too.